pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -119,28 +119,75 @@ func TestConfigMapControllerDisabling(t *testing.T) {
 		result              map[string][]string
 	}{
 		{
-			name: "ControllersEnabled",
+			name: "CapabilitiesEnabled",
 			knownCapabilities: []v1.ClusterVersionCapability{
 				configv1.ClusterVersionCapabilityBuild,
 				configv1.ClusterVersionCapabilityDeploymentConfig,
+				configv1.ClusterVersionCapabilityImageRegistry,
 			},
 			enabledCapabilities: []v1.ClusterVersionCapability{
 				configv1.ClusterVersionCapabilityBuild,
 				configv1.ClusterVersionCapabilityDeploymentConfig,
+				configv1.ClusterVersionCapabilityImageRegistry,
 			},
 			result: map[string][]string{"controllers": {"*"}},
 		},
 		{
-			name: "ControllersDisabled",
+			name: "BuildCapDisabled",
+			knownCapabilities: []v1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+			},
+			enabledCapabilities: []v1.ClusterVersionCapability{},
+			result: map[string][]string{
+				"controllers": {"*",
+					"-openshift.io/build",
+					"-openshift.io/build-config-change",
+					"-openshift.io/builder-rolebindings",
+				}},
+		},
+		{
+			name: "DeploymentConfigCapDisabled",
+			knownCapabilities: []v1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []v1.ClusterVersionCapability{},
+			result: map[string][]string{
+				"controllers": {"*",
+					"-openshift.io/deployer-rolebindings",
+					"-openshift.io/deploymentconfig",
+				}},
+		},
+		{
+			name: "ImageRegistryCapDisabled",
+			knownCapabilities: []v1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityImageRegistry,
+			},
+			enabledCapabilities: []v1.ClusterVersionCapability{},
+			result: map[string][]string{
+				"controllers": {"*",
+					"-openshift.io/image-puller-rolebindings",
+				}},
+		},
```

Comment:
/cc @flavianmissi 

Verify that the image puller roles should be disabled if ImageRegistry capability is diabled

Reply:
sounds good to me. do we want image registry QE to take this for a spin?

Reply:
https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/configobservation/controllers/capability_imageregistry.go includes `openshift.io/serviceaccount-pull-secrets` as well.